modules/siddhi-extensions/markov-models/src/main/java/org/wso2/siddhi/extension/markov/MarkovChainStreamProcessor.java

```diff
@@ -0,0 +1,280 @@
+     * @param executionPlanContext the context of the execution plan
+     * @return the additional output attributes introduced by the function
+     */
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+            ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+
+        if (attributeExpressionExecutors.length != 5 && attributeExpressionExecutors.length != 6) {
+            throw new ExecutionPlanValidationException(
+                    "Markov chain function has to have exactly 5 or 6 parameters, currently "
+                            + attributeExpressionExecutors.length + " parameters provided");
+        }
+
+        trainingOption = true;
+        trainingMode = TrainingMode.REAL_TIME;
+
+        if (!(attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor)) {
+            throw new ExecutionPlanValidationException("Duration has to be a constant");
+        }
+
+        if (!(attributeExpressionExecutors[3] instanceof ConstantExpressionExecutor)) {
+            throw new ExecutionPlanValidationException("Alert threshold probability value has to be a constant");
+        }
+
+        if (!(attributeExpressionExecutors[4] instanceof ConstantExpressionExecutor)) {
+            throw new ExecutionPlanValidationException("Training batch size has to be a constant");
+        }
+
+        Object durationObject = attributeExpressionExecutors[2].execute(null);
+        if (durationObject instanceof Integer) {
+            durationToKeep = (Integer) durationObject;
+        } else if (durationObject instanceof Long) {
+            durationToKeep = (Long) durationObject;
+        } else {
+            throw new ExecutionPlanValidationException("Duration should be of type int or long. But found "
+                    + attributeExpressionExecutors[2].getReturnType());
+        }
+
+        Object alertThresholdProbabilityObject = attributeExpressionExecutors[3].execute(null);
+        if (alertThresholdProbabilityObject instanceof Double) {
+            alertThresholdProbability = (Double) alertThresholdProbabilityObject;
+        } else {
+            throw new ExecutionPlanValidationException(
+                    "Alert threshold probability should be of type double. But found "
+                            + attributeExpressionExecutors[3].getReturnType());
+        }
+
+        Object object = attributeExpressionExecutors[4].execute(null);
+        if (object instanceof String) {
+            markovMatrixStorageLocation = (String) object;
+            trainingMode = TrainingMode.PREDEFINED_MATRIX;
+
+            File file = new File(markovMatrixStorageLocation);
+            if (!file.exists()) {
+                throw new ExecutionPlanValidationException(
+                        markovMatrixStorageLocation + " is not exists. Please provide a valid file path");
```

Comment:
Thanks for pointing it out. :) Fixed with https://github.com/wso2/siddhi/pull/189/commits/690b60924a459f505cce35f57d2ce6b05e4980c2